internal/secrets/gcpkms/kms.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limtations under the License.
+
+// Package gcpkms provides functionality to encrypt and decrypt secrets using
+// Google Cloud KMS.
+package gcpkms
+
+import (
+	"context"
+	"fmt"
+
+	cloudkms "cloud.google.com/go/kms/apiv1"
+	"github.com/google/go-cloud/gcp"
+	"google.golang.org/api/option"
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+)
+
+// endPoint is the address to access Google Cloud KMS API.
+const endPoint = "cloudkms.googleapis.com:443"
+
+// DefaultClient returns a client to use with Cloud KMS and a clean-up function
+// to close the client after used.
+func DefaultClient(ctx context.Context, ts gcp.TokenSource) (*cloudkms.KeyManagementClient, func(), error) {
+	c, err := cloudkms.NewKeyManagementClient(ctx, option.WithTokenSource(ts))
+	return c, func() { c.Close() }, err
+}
+
+// NewCrypter returns a new Crypter to to encryption and decryption.
+func NewCrypter(ki *KeyInfo, client *cloudkms.KeyManagementClient) *Crypter {
```

Comment:
nit: Flip the order of these parameters. Go from most general type to least general.

```suggestion
func NewCrypter(client *cloudkms.KeyManagementClient, ki *KeyInfo) *Crypter {
```